src/ripple/rpc/handlers/GatewayBalances.cpp

```diff
@@ -90,17 +86,7 @@ doGatewayBalances(RPC::JsonContext& context)
         auto addHotWallet = [&hotWallets](Json::Value const& j) {
             if (j.isString())
             {
-                auto const pk = parseBase58<PublicKey>(
-                    TokenType::AccountPublic, j.asString());
-                if (pk)
-                {
-                    hotWallets.insert(calcAccountID(*pk));
-                    return true;
-                }
-
-                auto const id = parseBase58<AccountID>(j.asString());
-
-                if (id)
+                if (auto const id = parseBase58<AccountID>(j.asString()); id)
```

Comment:
The original code is:
```
auto const id = parseBase58<AccountID>(j.asString());

                if (id)
                {
                    hotWallets.insert(*id);
                    return true;
                }
```
In this case it does need to be checked. In case of `if` with initializer it's checked if there is no other condition.